rust_src/build.rs

```diff
@@ -0,0 +1,206 @@
+use std::env;
+use std::fs;
+use std::fs::File;
+use std::io;
+use std::io::{BufRead, BufReader, Write};
+use std::path::{Path, PathBuf};
+
+#[derive(Eq, PartialEq)]
+enum ParseState {
+    Looking,
+    IgnoreComments,
+    NoMangleFn,
+    LispFn(Option<String>),
+}
+
+fn get_function_name(line: &str) -> Option<String> {
+    if let Some(pos) = line.find('(') {
+        if let Some(fnpos) = line.find("fn ") {
+            return Some((&line[(fnpos + 3)..pos]).to_string());
+        }
+    }
+
+    None
+}
+
+static C_NAME: &'static str = "c_name = \"";
+
+fn c_exports_for_module<R>(
+    modname: &str,
+    mut out_file: &File,
+    in_file: R,
+) -> Result<bool, io::Error>
+where
+    R: BufRead,
+{
+    let mut parse_state = ParseState::Looking;
+    let mut has_lisp_fns = false;
+    let mut exported: Vec<String> = Vec::new();
+    let mut has_include = false;
+
+    for line in in_file.lines() {
+        let line = line?;
+
+        match parse_state {
+            ParseState::Looking => {
+                if line.starts_with("#[lisp_fn") {
+                    if let Some(begin) = line.find(C_NAME) {
+                        let input = line.to_string();
```

Comment:
This string copy seems unnecessary.

Reply:
It was complaining about temporaries and borrowing if I recall.